Sources/Fuzzilli/Core/ProgramBuilder.swift

```diff
@@ -299,42 +303,160 @@ public class ProgramBuilder {
                 parameterTypes.append(varargsParam.removingFlagTypes())
             }
         }
-            
+
         for param in parameterTypes {
             if param.isOptional {
                 // It's an optional argument, so stop here in some cases
                 if probability(0.25) {
                     break
                 }
             }
-            
+
             assert(!param.isList)
-            guard let v = randVar(ofType: param) else { return nil }
-            arguments.append(v)
+            if !block(param, &arguments) {
+                return nil
+            }
         }
-            
+
         return arguments
     }
-    
-    public func generateCallArguments(for function: Variable) -> [Variable]? {
+
+    public func generateCallArguments(for signature: FunctionSignature) -> [Variable] {
+        return buildCallArgumentsInternal(for: signature, block: { param, arguments in
+            // Sometimes we should take the available variables here.
+            // TODO(cffsmith): should we just always generate in this function?
+            if let v = randVar(ofType: param), probability(0.5) {
```

Comment:
I feel like we only want to generate code here if we don't have a match in our current program (otherwise, we again create disjoint dataflows). But I think for that you'll need `randVar(ofConservativeType:)`?